main.go

```diff
@@ -0,0 +1,42 @@
+package main
+
+import (
+	"fmt"
+)
+
+// FirehoseBackend is used for consuming messages from a transport and read/write to storage
+type FirehoseBackend interface {
+}
+
+type FirehoseProcess struct {
```

Comment:
not sure about difference between `FirehoseProcess` and `Firehose` structs.

Reply:
I was sort of modeling it off this https://github.com/cloudchacho/hedwig-go/blob/main/consumer.go#L24, guessing process will have more in it later on?  I can probably just make it one too

Reply:
changed to one